dnf5/commands/history/history_store.cpp

```diff
@@ -19,14 +19,58 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 
 #include "history_store.hpp"
 
+#include "commands/history/transaction_id.hpp"
+
+#include <libdnf5/utils/bgettext/bgettext-mark-domain.h>
+
+#include <fstream>
+
 namespace dnf5 {
 
 using namespace libdnf5::cli;
 
 void HistoryStoreCommand::set_argument_parser() {
-    get_argument_parser_command()->set_description("Store transaction to a file");
+    auto & cmd = *get_argument_parser_command();
+    cmd.set_description("Store transaction to a file");
+    auto & ctx = get_context();
+    auto & parser = ctx.get_argument_parser();
+
+    output_option = dynamic_cast<libdnf5::OptionString *>(
+        parser.add_init_value(std::make_unique<libdnf5::OptionString>("./transaction.json")));
+    auto query_format = parser.add_new_named_arg("output");
+    query_format->set_long_name("output");
+    query_format->set_short_name('o');
+    query_format->set_description("File path for storing the transaction, default is \"./transaction.json\"");
+    query_format->set_has_value(true);
+    query_format->set_arg_value_help("PATH");
+    query_format->link_value(output_option);
+    cmd.register_named_arg(query_format);
+
+    transaction_specs = std::make_unique<TransactionSpecArguments>(*this);
 }
 
-void HistoryStoreCommand::run() {}
+void HistoryStoreCommand::run() {
+    const auto ts_specs = transaction_specs->get_value();
+    auto & history = *get_context().base.get_transaction_history();
+    std::vector<libdnf5::transaction::Transaction> transactions;
+
+    if (ts_specs.empty()) {
+        transactions = list_transactions_from_specs(history, {"last"});
+    } else {
+        transactions = list_transactions_from_specs(history, ts_specs);
+    }
+
+    //TODO(amatej): Allow storing multiple transactions at once?
+    if (transactions.size() != 1) {
+        throw libdnf5::cli::CommandExitError(1, M_("Multiple transactions selected for storing, only one allowed."));
+    }
+
+    std::filesystem::path path(output_option->get_value());
+    std::ofstream serialized_trans;
+    //TODO(amatej): add warning if file already exists?
+    serialized_trans.open(path);
+    serialized_trans << transactions[0].serialize_to_json();
+    serialized_trans.close();
```

Comment:
I would recommend to perform the operation atomically .
1. Create a temp file
2. Generate content an save it in temp unique file
3. Move temp file to the new location

This approach will ensure that the output will be 100% correct and not only partially written or corrupted.

Reply:
I have changed it, please take a look.